third_party/move/move-vm/types/src/loaded_data/runtime_types.rs

```diff
@@ -130,6 +129,12 @@ impl StructType {
 #[derive(Debug, Copy, Clone, Eq, Hash, Ord, PartialEq, PartialOrd)]
 pub struct CachedStructIndex(pub usize);
 
+#[derive(Debug, Clone, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct StructName {
+    pub module: ModuleId,
+    pub name: Identifier,
```

Comment:
Can't we make this be an index? i.e. the index of the struct inside the module's struct definitions? The real name, if we need it, can then be queried?
anyway, the compression on identifiers might help, but trying to understand why we need a string here.

Reply:
Yes a better compression on identifier would definitely help. right now I did a bit of compression by putting StructName behind an Arc so that each module will have a fixed number of Arcs. This will save lots of duplicated allocation already. However since we don't have globally unique identifier pool we still need to look into the full name for comparision/comupting hashes, which we could save if we have this compression pool.